test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -1304,11 +1304,17 @@ func deleteNodePool(name string) {
 	glog.Infof("Deleting node pool %s", name)
 	args := []string{"container", "node-pools", "delete", name, "--quiet",
 		"--cluster=" + framework.TestContext.CloudConfig.Cluster}
-	output, err := execCmd(getGcloudCommand(args)...).CombinedOutput()
-	if err != nil {
-		glog.Infof("Error: %v", err)
-	}
-	glog.Infof("Node-pool deletion output: %s", output)
+	wait.ExponentialBackoff(
+		wait.Backoff{Duration: 1 * time.Minute, Factor: float64(3), Steps: 3},
+		func() (bool, error) {
+			output, err := execCmd(getGcloudCommand(args)...).CombinedOutput()
+			if err != nil {
+				glog.Warningf("Error deleting nodegroup - error:%v, output: %s", err, output)
+				return false, nil
```

Comment:
nit: we should probably fail the test which triggered clean-up after the last attempt failed. This won't prevent disrupting subsequent tests, but if the same one is causing this problem across runs, at least we'll see the trend on the test-grid

Reply:
Makes sense, will do. Although currently in regional tests I'm seeing time-outs when this happens, as all the subsequent tests wait for a lower number of nodes to be available both at the beginning of the test (5 minutes) and at the end of the test (20 minutes). That's something to consider fixing too (at least skip the second wait if the first one fails), but next PR.

Reply:
Done.